ml-fundamentals/ml-feasibility-study.md

```diff
@@ -0,0 +1,98 @@
+
+## What do ML feasibility studies entail?
+
+### ML problem definition and desired outcome
+
+* Ensuring that the problem is complex enough that coding rules or manual scaling is unrealistic
+* Clear definition of the problem from the ML perspective
+* Definition of precisely what will the ML component solve
+
+### Data discovery
+
+* Hold a data exploration workshop and deep dive with domain experts
+* Understand the data dictionary, data quality and access (availability)
+* Ensure required data is present in reasonable volumes
+* For supervised problems (most common), assess the availability of labels or data that can be used to effectively approximate labels
+* If applicable, ensure all data can be joined as required and understand how
+  * Ideally obtain or create an entity relationship diagram (ERD)
+* Potentially uncover new useful data sources
+
+### Architecture discovery
+
+* Clear picture of existing architecture
+* Infrastructure spikes
+
+### Exploratory data analysis (EDA)
+
+* Data deep dive
+* Understand feature value distributions
+* Understand feature correlations
+* Understand data specific problem constraints like missing values, categorical cardinality, potential for data leakage etc.
+* Identify any gaps in data that couldn’t be identified in the data discovery phase
+* Pave the way of further understanding of what techniques are applicable
+* Establish a mutual understanding of what data is in or out of scope for feasibility, ensuring that the data in scope is significant for the business
+
+### Data pre-processing
+
+* Happens during EDA and hypothesis testing
+* Feature engineering
+* Sampling
+* Scaling and/or discretization
+* Noise handling
+
+### Hypothesis testing
+
+* Design several potential solutions using theoretically applicable algorithms and techniques
+* Train model(s)
+* Evaluate performance and determine if satisfactory
+* Tweak experimental solution designs based on outcomes
+* Iterate
+  * Thoroughly document each step and outcome, plus any resulting hypotheses for easy following of the decision making process
+
+### Risk assessment
+
+* Identification and assessment of risks and constraints from ML standpoint
+
+### Responsibile AI
+
+* Consideration of responsible AI and fairness
+* Discussion and feedback from diverse perspectives around any resposible AI concerns
+
+## Output of a ML feasibilty study
+
+### Possible outcomes
+
+If this process does not produce sufficient evidence, aligned to the pre-determined performance measures and business impact, to support the hypothesis that this problem can be solved using ML
```

Comment:
```suggestion
If there is not enough evidence to support the hypothesis that this problem can be solved using ML
```

Reply:
the business alignment part was suggested by one of our leaders in order to emphasize it's importance. I've re-arranged the sentence to something that I hope is more readable:
If there is not enough evidence to support the hypothesis that this problem can be solved using ML, as aligned with the pre-determined performance measures and business impact